### PR TITLE
Tests for recent C fixes

### PR DESCRIPTION
**Pull request by eafer**
These are the updated test schemas for the recently merged C generator patches. As requested by @wing328 here:

  https://github.com/OpenAPITools/openapi-generator/pull/14434#issuecomment-2497497110

### CODE REVIEW

**Comment by eafer** [comment]
@ityuhui

File: modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml [hunk 1]
```diff
@@ -626,6 +644,17 @@ definitions:
         type: string
     xml:
       name: Category
+  Preference:
```
[the remark this answers is not in the document]
Comment (eafer) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: What new files? Did I mess something up? I thought I just needed to update the schema with the new cases that were failing.

[the remark this answers is not in the document]
Comment (eafer) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: Oh, you mean for the samples? Yes, I added everything.

[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: from the build error message:
```
[ 66%] Building C object CMakeFiles/openapi_petstore.dir/model/tag.c.o
[ 73%] Building C object CMakeFiles/openapi_petstore.dir/model/user.c.o
In file included from /home/runner/work/openapi-generator/openapi-generator/samples/client/petstore/c/model/user.c:4:
/home/runner/work/openapi-generator/openapi-generator/samples/client/petstore/c/model/user.h:19:10: fatal error: preference.h: No such file or directory
   19 | #include "preference.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/openapi_petstore.dir/build.make:219: CMakeFiles/openapi_petstore.dir/model/user.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/openapi_petstore.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

looks like the c petstore like is trying to import `preference` (enum) but no luck.

does the C petstore client support enum models? or it only supports inline enum?

[the remark this answers is not in the document]
Comment (eafer) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: You are right, I messed with up at some point during rebasing. I'll send the rest of the samples now.

[the remark this answers is not in the document]
Comment (eafer) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: I just added the missing files, sorry for the confusion.

[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: https://github.com/OpenAPITools/openapi-generator/actions/runs/12054702867/job/33613548050?pr=20200

can you please update the samples one more time?

**Comment by wing328** [comment]
cc @zhemant (2018/11) @ityuhui (2019/12) @michelealbano (2020/03)

File: modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wing328) [reply modules/openapi-generator/src/test/resources/2_0/c/petstore.yaml:647]: No problem bro. Thanks for adding the test.

**Comment by wing328** [comment]
lgtm. let's give it a try

**Pull request merged**